Source/Events.Store.MongoDB/EventHorizon/EventHorizonEventsWriter.cs

```diff
@@ -18,47 +17,40 @@ namespace Dolittle.Runtime.Events.Store.MongoDB.EventHorizon
     public class EventHorizonEventsWriter : IWriteEventHorizonEvents
     {
         readonly FilterDefinitionBuilder<MongoDB.Events.Event> _eventFilter = Builders<MongoDB.Events.Event>.Filter;
-        readonly EventStoreConnection _connection;
+        readonly IStreams _streams;
+        readonly IEventsToStreamsWriter _eventsToStreamsWriter;
+        readonly IEventConverter _eventConverter;
         readonly ILogger _logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="EventHorizonEventsWriter"/> class.
         /// </summary>
-        /// <param name="connection">An <see cref="EventStoreConnection"/> to a MongoDB EventStore.</param>
-        /// <param name="logger">An <see cref="ILogger"/>.</param>
-        public EventHorizonEventsWriter(
-            EventStoreConnection connection,
-            ILogger logger)
+        /// <param name="streams">The <see cref="IStreams" />.</param>
+        /// <param name="eventsToStreamsWriter">The <see cref="IEventsToStreamsWriter" />.</param>
+        /// <param name="eventConverter">The <see cref="IEventConverter" />.</param>
+        /// <param name="logger">The <see cref="ILogger" />.</param>
+        public EventHorizonEventsWriter(IStreams streams, IEventsToStreamsWriter eventsToStreamsWriter, IEventConverter eventConverter, ILogger logger)
```

Comment:
Same thing here.

Also `IWriteEventHorizonEvents` doesn't mention anything about `Streams`, should it or is the `Streams` part always implicit in which case we could drop it from the other `IWriteXToY` interfaces?
```suggestion
        /// <param name="eventsToStreamsWriter">The <see cref="IWriteStoreEventsToStreams" />.</param>
        /// <param name="eventConverter">The <see cref="IEventConverter" />.</param>
        /// <param name="logger">The <see cref="ILogger" />.</param>
        public EventHorizonEventsWriter(IStreams streams, IWriteStoreEventsToStreams eventsToStreamsWriter, IEventConverter eventConverter, ILogger logger)
```

Reply:
```suggestion
        /// <param name="eventsToStreamsWriter">The <see cref="IWriteEventsToStreamCollection" />.</param>
        /// <param name="eventConverter">The <see cref="IEventConverter" />.</param>
        /// <param name="logger">The <see cref="ILogger" />.</param>
        public EventHorizonEventsWriter(IStreams streams, IWriteEventsToStreamCollection eventsToStreamsWriter, IEventConverter eventConverter, ILogger logger)
```